ui/app/adapters/pki/issuer.js

```diff
@@ -39,13 +41,42 @@ export default class PkiIssuerAdapter extends ApplicationAdapter {
     return this.ajax(url, 'POST', { data });
   }
 
-  query(store, type, query) {
-    return this.ajax(this.urlForQuery(query.backend), 'GET', this.optionsForQuery());
+  async query(store, type, query) {
+    const { backend } = query;
+    const url = this.urlForQuery(backend);
+
+    return this.ajax(url, 'GET', this.optionsForQuery()).then(async (res) => {
+      if (res.data.keys.length <= 10) {
+        const records = await all(
+          res.data.keys.map((id) => {
+            return this.queryRecord(store, type, { id, backend });
+          })
+        );
+
+        const isRootRecords = await all(
+          records.map((record) => {
+            const { certificate } = record.data;
+            const keyId = record.data.key_id;
+
+            return verifyCertificates(certificate, certificate) && !!keyId;
```

Comment:
nit: I think switching the order here would mean that if `!keyId` then we wouldn't even bother verifying the certificates since it's an AND operator
```suggestion
            return !!keyId && verifyCertificates(certificate, certificate);
```